src/test/java/org/apache/commons/imaging/common/bytesource/ByteSourceTest.java

```diff
@@ -66,6 +64,22 @@ protected File createTempFile(final byte src[]) throws IOException {
         }
         final byte longArray[] = (baos.toByteArray());
 
-        return new byte[][] { emptyArray, single, simple, zeroes, longArray, };
+        return new byte[][]{emptyArray, single, simple, zeroes, longArray,};
+    }
+
+    @Test
```

Comment:
@onealj @TheRealHaui If I were maintaining this code, I'd be happy with either pattern.

`@Test(expected=NullPointerException.class)` would be a little shorter and equivalent to the `try..catch` construct used here in this case.

However, the reason people tend to prefer `try..catch` is because in a lot of cases it is not granular enough. Take this (admittedly contrived) example:

```java
@Test(expected=NullPointerException.class)
public void doSomething() {
  String s = null;
  assertTrue(s.isEmpty());
  try {
    somethingWhichThrowsNullPointerException():
  } catch (NullPointerException e) {
    assertNull(s);
  }
}
```

In this case, rather than `somethingWhichThrowsNullPointerException(s1)` throwing `NullPointerException` (which we expect), it will be `s.isEmpty()` that throws it instead, which isn't what we intended, since it means the test will always "pass".